configurate-core/src/main/java/ninja/leaping/configurate/objectmapping/serialize/TypeSerializers.java

```diff
@@ -91,6 +92,20 @@ public void serialize(@NonNull TypeToken<?> type, @Nullable String obj, @NonNull
     }
 
     private static class NumberSerializer implements TypeSerializer<Number> {
+
+        public static Predicate<TypeToken<Number>> getPredicate() {
+            return (type) -> {
+                type = type.wrap();
+                Class<?> clazz = type.getRawType();
+                return Integer.class.equals(clazz)
+                        || Long.class.equals(clazz)
+                        || Short.class.equals(clazz)
+                        || Byte.class.equals(clazz)
+                        || Float.class.equals(clazz)
+                        || Double.class.equals(clazz);
```

Comment:
I would have expected more types then this being supported by default. BigDecimal?

Reply:
My intent with this PR is just to fix the behavior of the existing serialized. We could probably add support for BigDecimal in another PR